---
title: "CDC / Johns Hopkins Technical Guide to Getting Started with MATLINK"
date: "July 8, 2024"
toc: true
toc-title: "On this page"
---

# Welcome {#welcome}

## INTRO VIDEO, KHYZER AND PAUL: NEED A PLACE TO STORE THE VIDEO -- TOO LARGE FOR GITHUB

## DIAGRAM OF MATLINK program?

## Background

Miele K, Kim SY, Jones R, et al. Medication for Opioid Use Disorder During Pregnancy — Maternal and Infant Network to Understand Outcomes Associated with Use of Medication for Opioid Use Disorder During Pregnancy (MAT-LINK), 2014–2021. MMWR Surveill Summ 2023;72(No. SS-3):1–14. [DOI: http://dx.doi.org/10.15585/mmwr.ss7203a1](http://dx.doi.org/10.15585/mmwr.ss7203a1). <https://www.cdc.gov/mmwr/volumes/72/ss/ss7203a1.htm>

## Meeting schedule {#meeting-schedule}

## REdCAP Training Resouces

[REDCap training videos](https://projectredcap.org/resources/videos/)

## MATLINK-specific REDCap resources, FAQs - TBD

TO BE INCLUDED

## Overview of Major Clinical Terminologies and Coding Systems {#overview-of-major-clinical-terminologies-and-coding-systems}

# Learning to Use EHR Data in Learning Health Systems

![Dr. Rachel Richesson presents "Learning to Use EHR Data in Learning Health Systems"](https://img.youtube.com/vi/4uXqNLsiVuc/maxresdefault.jpg)

"Computable phenotypes can facilitate research and learning by supporting the identification of patient populations, the delivery of clinical interventions, and the assessment of outcomes. The sharing and reuse of computable phenotypes can enhance the efficiency of pragmatic research and the dissemination of evidence-based interventions into real-world settings. This talk will discuss current platforms for identifying existing computational phenotypes, as well as challenges and strategies for their implementation and validation in learning health systems." 

This event was originally presented on February 9, 2023, as part of the webinar series: Using Health System Research to Revolutionize Rehabilitation Care.

[Watch the video on YouTube](https://youtu.be/4uXqNLsiVuc?feature=shared)


This section provides a detailed overview of several essential clinical terminologies and coding systems used in healthcare. Each system has a specific role and is crucial for standardized communication in healthcare settings. The information includes development history, usage, and updates of these systems.

### SNOMED Clinical Terms (SNOMED CT) {#snomed-clinical-terms-snomed-ct}

- **Origin:** Developed by the College of American Pathologists, now under SNOMED International.
- **Usage:** Used in more than 50 countries to encode clinical information, including diseases, findings, and procedures.
- **Scope:** More than 340,000 active concepts in 19 hierarchies.
- **Updates:** Biannual, with more frequent updates planned.
- **More Information:** [SNOMED International](https://www.snomed.org/snomed-ct)

### Logical Observation Identifiers Names and Codes (LOINC) {#logical-observation-identifiers-names-and-codes-loinc}

- **Origin:** Developed by the Regenstrief Institute.
- **Usage:** Provides identifiers for laboratory and clinical observations.
- **Scope:** Over 90,000 terms.
- **Updates:** Biannual, in collaboration with SNOMED CT for coded content development.
- **More Information:** [LOINC](https://loinc.org)

### RxNorm {#rxnorm}

- **Origin:** Developed by the National Library of Medicine (NLM).
- **Usage:** Standard nomenclature for medications, linking to various drug vocabularies.
- **Access:** Requires UMLS user license for proprietary content.
- **More Information:** [RxNorm - NLM](https://www.nlm.nih.gov/research/umls/rxnorm)

### International Classification of Disease (ICD) {#international-classification-of-disease-icd}

- **Origin:** Endorsed by the World Health Organization (WHO).
- **Usage:** Widely used for epidemiology, health management, and clinical purposes.
- **Scope:** ICD-10 is widely used with national extensions; ICD-11 is adopted for future use.
- **Updates:** Annual, freely available.
- **More Information:** [WHO ICD](https://www.who.int/classifications/icd)

### Current Procedural Terminology (CPT) {#current-procedural-terminology-cpt}

- **Origin:** Developed by the American Medical Association (AMA).
- **Usage:** Encoding of medical services and procedures in the USA.
- **Scope:** Contains three categories of codes.
- **Access:** Requires a license from AMA for use.
- **More Information:** [CPT - AMA](https://www.ama-assn.org/practice-management/cpt)

### Human Phenotype Ontology (HPO) {#human-phenotype-ontology-hpo}

- **Origin:** Developed as a bioinformatics resource for human diseases and phenotypes analysis.
- **Usage:** Used in genomic interpretation, gene-disease discovery, and precision medicine.
- **Scope:** Contains more than 13,000 terms in 5 hierarchies.
- **Updates:** Freely available, with multiple releases per year.
- **More Information:** [Human Phenotype Ontology](https://hpo.jax.org)

### Unified Medical Language System (UMLS) {#unified-medical-language-system-umls}

- **Origin:** Developed by the US National Library of Medicine in 1986.
- **Usage:** Facilitates the retrieval and integration of electronic biomedical information.
- **Scope:** Addresses different vocabularies expressing the same information differently.
- **Access:** Free, but requires a license due to additional licensing requirements of some contents.
- **More Information:** [UMLS - NLM](https://www.nlm.nih.gov/research/umls)


### Simple Standard for Sharing Ontological Mappings (SSSOM) {#simple-standard-for-sharing-ontological-mappings-sssom}

- **Origin:** Developed by the scientific community as a lightweight standard for sharing mappings between ontologies.
- **Usage:** Facilitates the exchange and interoperability of mapping data between different ontologies and vocabularies.
- **Scope:** Supports a variety of mapping types and includes metadata to describe mappings.
- **Updates:** Developed collaboratively with input from various stakeholders to ensure broad applicability and utility.
- **More Information:** [SSSOM GitHub Repository](https://github.com/mapping-commons/sssom)

### Mondo Disease Ontology (MONDO) {#mondo-disease-ontology-mondo}

- **Origin:** Developed by the Monarch Initiative to harmonize disease definitions across resources.
- **Usage:** Provides a unified disease ontology to support data integration and interoperability.
- **Scope:** Integrates multiple disease resources into a single ontology with more than 20,000 terms.
- **Updates:** Regularly updated with contributions from the community.
- **More Information:** [MONDO GitHub Repository](https://github.com/monarch-initiative/mondo)
```

    ::: {style="text-align: right"}
    <a href="#table-of-contents">Back to Table of Contents</a>
    :::

https://www.healthit.gov/isa/section-i-vocabularycode-setterminology-standards-and-implementation-specifications

## OMOP Domains by Source to Standard Vocabulary {#omop-domains-by-source-to-standard-vocabulary}

``` mermaid
graph LR
    ICD9("ICD9") -->|Transformation to OMOP CDM| SNOMED("STANDARD<br>Vocabulary Concept Code<br>SNOMED")
    ICD10("ICD10") -->|Transformation to OMOP CDM| SNOMED
```

::: {style="text-align: right"}
<a href="#table-of-contents">Back to Table of Contents</a>
:::

| Domain       | Source Vocabulary                     | Standard Vocabulary |
|--------------|---------------------------------------|---------------------|
| Conditions   | ICD9, ICD10                           | SNOMED              |
| Measurements | LOINC or institutional specific codes | LOINC               |
| Drugs        | NDC                                   | RxNORM              |
| Procedures   | ICD9, ICD10, CPT                      | SNOMED              |

-   ICD = International Classification of Diseases

-   SNOMED = Systematized Nomenclature of Medicine

-   LOINC = Logical Observation Identifiers Names and Codes

-   NDC = National Drug Code

-   CPT = Current Procedural Terminology

    ::: {style="text-align: right"}
    <a href="#table-of-contents">Back to Table of Contents</a>
    :::

## Welcome to OMOP for MATLINK users {#welcome-to-omop-for-matlink-users}

This site is designed to provide guidance, scripts and resources for sites transitioning data to OMOP common data model as part of MATLINK.

MATLINK is a public health surveillance system designed and developed by the Centers for Disease Control and Prevention (CDC) to monitor maternal, infant, and child health outcomes associated with medication for opioid use disorder during pregnancy.

## Introduction {#introduction}

The goals of OMOP are to gain better understanding of human health by improving our ability to analyze observational health data.

Health data is stored by individual health systems in unique ways which limits the ability to collaborate and learn from larger populations. This is a problem because certain clinical conditions are infrequent and it is important to have sufficient number of cases to perform statistical tests. De-identified data shared across institutions has the promise of allowing scientists to develop important insights about health which is the motivation to enable greater collaboration.

#### What is a common data model? {#what-is-a-common-data-model}

A common data model (CDM) allows multiple health care systems with their individual databases to join forces creating greater populations which can power more compelling scientific studies. A simple way to think about this is the difficulty of finding the silverware drawer in a new kitchen -- if there is a unifying rule for where the silverware drawer is located in every kitchen (immediately to right of the sink) it makes it possible to find this much easier. A common data model creates a system so that data elements like "systolic blood pressure" or "sepsis" are recorded and organized in the same way. OMOP stands for Observational Medical Outcomes Partnership and is the common data model of the [Observational Health Data Sciences and Informatics](https://www.ohdsi.org/) program.

#### What is an ETL? {#what-is-an-etl}

ETL stands for Extract, Transform, and Load. Essentially this is the process of mapping data from one system to another. Please see \[The Book of OHDSI, Chapter 6\] (https://ohdsi.github.io/TheBookOfOhdsi/ExtractTransformLoad.html).

## Getting started {#getting-started}

The code for the MATLINK project is hosted on \[GitHub\]XXXX

## MATLINK technical support checklist {#matlink-technical-support-checklist}

1.  **TBD**

### [OHDSI ETL Resources]{.underline} {#ohdsi-etl-resources}

-   [OHDSI Wiki: ETL Creation Best Practices](https://www.ohdsi.org/web/wiki/doku.php?id=documentation:etl_best_practices)
-   [Book of OHDSI (Ch. 6): Extract Transform Load](Book%20of%20OHDSI%20-%20Ch.6%20Extract%20Transform%20Load)
-   [EHDEN Academy: OMOP CDM and Standardised Vocabularies](https://academy.ehden.eu/enrol/index.php?id=4)
-   [EHDEN Academy: Extract, Transform and Load](https://academy.ehden.eu/course/view.php?id=7)
-   [EHDEN Academy: Introduction to Usagi & Code Mappings for an ETL](https://academy.ehden.eu/enrol/index.php?id=18)

### [OHDSI ETL Steps]{.underline} {#ohdsi-etl-steps}

1.  **Find your people.**

    a.  Assemble ETL team: both data experts, medical experts, CDM/vocabulary experts
    b.  Assign persons with medical knowledge to create code mappings
    c.  Assign persons with technical knowledge to implement the ETL
    d.  Schedule regular meetings for quality control maintenance

2.  **Understand your data.**

    a.  Take inventory of source data (tools: [White Rabbit](https://github.com/OHDSI/WhiteRabbit))
        i.  List of tables in database
        ii. List fields in each table
        iii. List distinct values of each field
        iv. Summarize the frequency of each value of each field.
    b.  Define relationships between source data tables and fields (tools: [Rabbit-in-a-Hat](https://github.com/OHDSI/WhiteRabbit))
        i.  Record both relationship definition decisions and reasoning behind these decisions
    c.  Determine standardized vocabularies that already exist in your source data (ICD-10, CPT, HCPCS, LOINC, etc.)
        i.  Many commonly used standardized coding systems have already been mapped to the OMOP vocabulary -- utilize the work done by others across the OHDSI community to accelerate the process of mapping your data systems to OMOP.
    d.  Determine the coding systems in your source data that are proprietary or not already mapped to OMOP (proprietary patient/visit identifiers, account numbers, charge codes, etc.)

3.  **Map your codes to OMOP concepts.**

    a.  Summarize the frequency of each code from your source data code sets that will require mapping to OMOP concepts.
    b.  Create your source data to OMOP mapping. (tools: [Usagi](https://github.com/OHDSI/Usagi))
        i.  Assign this task to the ETL team members with appropriate medical knowledge to discern which codes are most synonymous based on their descriptions. Medical knowledge is key for this step as semantic understanding of clinical descriptions is required to make decisions on mapping source data concepts to OMOP concepts.
        ii. Start with the most frequently used codes and determine your threshold of code frequency to include in the mapping.
        iii. Focus on a particular project or clinical domain to limit the scope of data needed for capture and conversion to OMOP.
        iv. Utilize the Usagi tool for mapping suggestions and searching based on similarity of code descriptions.

4.  **OMOPify your data!**

    a.  Determine the technology and software used or approved at your site for data storage, querying, and transfer. Determine the tools and technologies that team members and internal staff have expertise in.
    b.  Generate the OMOP DDLs
        i.  Using R: https://github.com/OHDSI/CommonDataModel
        ii. Using SQL: https://github.com/OHDSI/CommonDataModel/tree/v5.4.0/inst/ddl/5.4
    c.  Implement the ETL.
        i.  Assign this task to the ETL team members with technical knowledge and permission for accessing and extracting source data and writing to a database.
        ii. Many tools and technologies can be used for this step. Utilize the existing tools, technologies, and expertise of your ETL team and internal staff.

5.  **Evaluate your data.**

    a.  Involve everyone in the evaluation and maintenance of ETL and data quality.
    b.  Review the ETL design documentation, computer code used in implementation, and concept mappings
    c.  Perform a manual audit using a sample of patients from both source data and derived OMOP data
    d.  Compare summary level counts of key fields between source data and OMOP derived data. (tools: Achilles)

**★ Ask for help along the way!**

The MATLINK support team is here to help you. We also encourage you to connect with the OHDSI community to find others with experience, expertise, and guidance for each step of your ETL to OMOP journey. The open, collaborative community of OHDSI is the most powerful tool at your disposal. Use it (us)!

### [Places to Connect]{.underline}

-   OHDSI.org: https://www.ohdsi.org/

-   OHDSI Wiki: https://www.ohdsi.org/web/wiki/doku.php

-   OHDSI Forums: https://forums.ohdsi.org/

-   OHDSI Workgroups: https://www.ohdsi.org/workgroups/ \`\`\`

## OMOP Table and Field Basics {#omop-table-and-field-basics}

Adapted from OHDSI CDM Site: [Data Model Conventions](https://ohdsi.github.io/CommonDataModel/dataModelConventions.html)

#### Tables {#tables}

For the tables of the main domains of the CDM it is imperative that concepts used are strictly limited to the domain. For example, the CONDITION_OCCURRENCE table contains only information about conditions (diagnoses, signs, symptoms), but no information about procedures. Not all source coding schemes adhere to such rules. For example, ICD-9-CM codes, which contain mostly diagnoses of human disease, also contain information about the status of patients having received a procedure. The ICD-9-CM code V20.3 'Newborn health supervision' defines a continuous procedure and is therefore stored in the PROCEDURE_OCCURRENCE table.

#### Fields {#fields}

Variable names across all tables follow one convention:

|                     |                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
|------------|------------------------------------------------------------|
| **Notation**        | **Description**                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| \_SOURCE_VALUE      | Verbatim information from the source data, typically used in ETL to map to CONCEPT_ID, and not to be used by any standard analytics. For example, CONDITION_SOURCE_VALUE = '787.02' was the ICD-9 code captured as a diagnosis from the administrative claim.                                                                                                                                                                                     |
| \_ID                | Unique identifiers for key entities, which can serve as foreign keys to establish relationships across entities. For example, PERSON_ID uniquely identifies each individual. VISIT_OCCURRENCE_ID uniquely identifies a PERSON encounter at a point of care.                                                                                                                                                                                       |
| \_CONCEPT_ID        | Foreign key into the Standardized Vocabularies (i.e. the standard_concept attribute for the corresponding term is true), which serves as the primary basis for all standardized analytics. For example, CONDITION_CONCEPT_ID = [31967](http://athena.ohdsi.org/search-terms/terms/31967) contains the reference value for the SNOMED concept of 'Nausea'                                                                                          |
| \_SOURCE_CONCEPT_ID | Foreign key into the Standardized Vocabularies representing the concept and terminology used in the source data, when applicable. For example, CONDITION_SOURCE_CONCEPT_ID = [45431665](http://athena.ohdsi.org/search-terms/terms/45431665) denotes the concept of 'Nausea' in the Read terminology; the analogous CONDITION_CONCEPT_ID might be 31967, since SNOMED-CT is the Standardized Vocabulary for most clinical diagnoses and findings. |
| \_TYPE_CONCEPT_ID   | Delineates the origin of the source information, standardized within the Standardized Vocabularies. For example, DRUG_TYPE_CONCEPT_ID can allow analysts to discriminate between 'Pharmacy dispensing' and 'Prescription written                                                                                                                                                                                                                  |

For more information, see:

[Data Model Conventions](https://ohdsi.github.io/CommonDataModel/dataModelConventions.html#Data_Model_Conventions)

[How to Calculate Drug Dose](https://ohdsi.github.io/CommonDataModel/drug_dose.html)

[Clinical Data Tables](https://ohdsi.github.io/CommonDataModel/cdm53.html#Clinical_Data_Tables)

# OHDSI Tool Suite {#ohdsi-tool-suite}

The Edge Tool Suite are a set of OHDSI tools that should be deployed by the site that provide value around the OMOP CDM. This work was funded by the MATLINK initiative <https://cure.ncats.io>

The OHDSI open source software configured for deployment include:

-   The Atlas data science platform

-   The WebAPI backed for Atlas

-   The HADES statistical analysis packages

-   The Data Quality Dashboard

-   The Perseus ETL management system

#### Ways to deploy the software {#ways-to-deploy-the-software}

1.  Cloud vendor software configured for use.

[OHDSI on Azure (Includes Perseus, Atlas, and Hades)](https://github.com/microsoft/OHDSIonAzure)

[OHDSI on AWS (Includes Atlas and Hades)](https://github.com/OHDSI/OHDSIonAWS)

2.  Broadsea provides a set of docker containers that ease the cost of implementation

[Broadsea (Includes Atlas and Hades)](https://github.com/OHDSI/Broadsea)

3.  Sites can compile the tools from the source repositories

#### OHDSI Specific {#ohdsi-specific}

<https://github.com/OHDSI/CommonDataModel>

<https://github.com/OHDSI/Broadsea>

<https://github.com/OHDSI/Athena>

#### The ETL Process {#the-etl-process}

<https://github.com/OHDSI/Perseus>

<https://github.com/OHDSI/WhiteRabbit>

<https://github.com/OHDSI/Usagi>

#### ATLAS and Cohort Discovery {#atlas-and-cohort-discovery}

<https://github.com/OHDSI/Atlas>

<https://github.com/OHDSI/WebAPI>

#### Broadsea {#broadsea}

<https://www.youtube.com/watch?v=a9ZJURNRbUg>

#### Data Analysis {#data-analysis}

<https://github.com/OHDSI/Achilles>

<https://github.com/OHDSI/Hades>

<https://github.com/OHDSI/DataQualityDashboard>

# OMOP Scripts for Epic {#omop-scripts-for-epic}

["The Spectrum Code"](https://userweb.epic.com/User/788dbdff-6712-4b16-aabc-98cdb1be4ff9): scripts and documentation created by Roger Carlson at Corewell Health

[RUMC OMOP transition scripts from Clarity, Caboodle](https://userweb.epic.com/User/788dbdff-6712-4b16-aabc-98cdb1be4ff9): Code created by Rush University Medical Center

# Using synthetic data {#using-synthetic-data}

[MIMIC-IV](https://physionet.org/content/mimic-iv-demo-omop/0.9/1_omop_data_csv/#files-panel) 100-patient demo dataset based on MIMIC to create a OMOP instance

[Synthea](https://github.com/OHDSI/ETL-Synthea) GitHub site with files to use the synthetic patient generator Synthea

[Eunomia](https://ohdsi.github.io/Eunomia/) LInk to the Eunomia GitHub.io site with instructions and standard dataset files

# Useful Resources {#useful-resources}

[Web site for the OHDSI community](https://www.ohdsi.org/): http://www.ohdsi.org

[The Book of OHDSI:](https://ohdsi.github.io/TheBookOfOhdsi/) What is OHDSI and why should I care?

[EHDEN Academy](https://academy.ehden.eu/) : EHDEN Academy is a site with courses for developing skills working with OHDSI

[Tutorials & Workshops](https://www.youtube.com/playlist?list=PLpzbqK7kvfeXRQktX0PV-cRpb3EFA2e7Z) : Tutorial sessions 1-8 provide a comprehensive overview from vocabularies and creating cohorts to prediction

[OHDSI Forums](https://forums.ohdsi.org/) : Searchable, active user community

[OHDSI Community Dashboard](https://dash.ohdsi.org/) : Tracks publications, citations, researchers and activity within the OHDSI community

[Clinical Registry Efforts Within OHDSI (Sept. 13 Community Call)](https://www.youtube.com/watch?v=HhQ9x3bsv4o): Video discussing Perseus and Broadsea

[OHDSI-in-a-Box on GitHub](https://github.com/OHDSI/OHDSI-in-a-Box) : A learning environment created for the OHDSI community

[Integrating Flowsheet Data in OMOP Common Data Model for Clinical Research](https://arxiv.org/ftp/arxiv/papers/2109/2109.08235.pdf) Paper written by informatics teams at Stanford University and The Hospital for Sick Children

[Guide to privacy issues in OMOP journey](https://ohdsi.github.io/CommonDataModel/cdmPrivacy.html): Article outlining how to manage protected health information

[DQD](https://github.com/OHDSI/DataQualityDashboard) GitHub repository for Data Quality Dashboard tool

[PheKB](https://phekb.org/phenotypes) A phenotype repository

[Analyze observational patient data by using OHDSI with the OMOP CDM](https://learn.microsoft.com/en-us/azure/architecture/example-scenario/digital-health/patient-data-ohdsi-omop-cdm) Microsoft guide to OMOP CDM

[OHDSI on Azure GitHub](https://github.com/microsoft/OHDSIonAzure) Automation code and documentation for deploying OHDSI CDM in Azure

[How to develop capacity for observational research within a health system](https://www.ohdsi.org/2022showcase-79/) Presentation on building capacity from 2022 OHDSI Collaborator Showcase
